cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/store/DefaultStoreTest.java

```diff
@@ -517,6 +518,88 @@ public void testUpdateChangedApplication() throws ConflictException {
     Assert.assertEquals(FooMapReduceJob.class.getName(), spec.getMapReduce().get("mrJob3").getClassName());
   }
 
+  @Test
+  public void testAddApplicationWithoutMarkingLatest()
+      throws ConflictException {
+    ApplicationId appId = new ApplicationId("account1", "app1");
+    ApplicationMeta appMeta = new ApplicationMeta("app1", Specifications.from(new FooApp()),
+        new ChangeDetail(null, null, null,
+            System.currentTimeMillis()), null, false);
+    store.addApplication(appId, appMeta);
+
+    ApplicationMeta storedMeta = store.getApplicationMetadata(appId);
+    Assert.assertNotNull(storedMeta);
```

Comment:
assertNotNull is not good enough, it should be checking the actual data.

Reply:
similar comment for all the other tests

Reply:
removed the null checks and added checks to verify the actual metadata.